cnd/src/network/comit_ln.rs

```diff
@@ -577,3 +577,182 @@ impl NetworkBehaviourEventProcess<oneshot_behaviour::OutEvent<finalize::Message>
+            address_hint: Some(bob_addr.clone()),
+        };
+
+        let ethereum_id = EthereumIdentity::from(Address::default());
+
+        let lightning_id = {
+            let secp_pubkey = secp256k1::PublicKey::from_str(
+                "02c2a8efce029526d364c2cf39d89e3cdda05e5df7b2cbfc098b4e3d02b70b5275",
+            )
+            .unwrap();
+            lightning::PublicKey::from(secp_pubkey)
+        };
+
+        let alice_swap_params = HanEtherereumHalightBitcoinCreateSwapParams {
+            role: Role::Alice,
+            peer: bob_dial_info.clone(),
+            ethereum_identity: ethereum_id,
+            ethereum_absolute_expiry: Timestamp::from(10),
+            ethereum_amount: asset::Ether::zero(),
+            lightning_identity: lightning_id,
+            lightning_cltv_expiry: Timestamp::from(10),
+            lightning_amount: asset::Lightning::from_sat(0),
+        };
+
+        let bob_swap_params = HanEtherereumHalightBitcoinCreateSwapParams {
+            role: Role::Bob,
+            peer: alice_dial_info.clone(),
+            ethereum_identity: ethereum_id,
+            ethereum_absolute_expiry: Timestamp::from(10),
+            ethereum_amount: asset::Ether::zero(),
+            lightning_identity: lightning_id,
+            lightning_cltv_expiry: Timestamp::from(10),
+            lightning_amount: asset::Lightning::from_sat(0),
+        };
+
+        let alice_node_id = NodeLocalSwapId::from(Uuid::new_v4());
+        let bob_node_id = NodeLocalSwapId::from(Uuid::new_v4());
+
+        alice_swarm.initiate_communication(alice_node_id, alice_swap_params.clone());
+
+        bob_swarm.initiate_communication(bob_node_id, bob_swap_params.clone());
+
+        let (alice_event, bob_event) = runtime.block_on(async move {
+            let alice_fut = alice_swarm.next();
+            pin_mut!(alice_fut);
+            let bob_fut = bob_swarm.next();
+            pin_mut!(bob_fut);
+            join!(alice_fut, bob_fut)
+        });
+
+        match (alice_event, bob_event) {
+            (
+                BehaviourOutEvent::SwapFinalized {
+                    local_swap_id: alice_local_swap_id,
+                    swap_params: alice_swap_params,
+                    secret_hash: alice_secret_hash,
+                    ethereum_identity: _alice_ethereum_identity,
+                },
+                BehaviourOutEvent::SwapFinalized {
+                    local_swap_id: bob_local_swap_id,
+                    swap_params: bob_swap_params,
+                    secret_hash: bob_secret_hash,
+                    ethereum_identity: _bob_ethereum_id,
+                },
+            ) => {
+                assert_eq!(alice_swap_params.peer.peer_id, bob_peer_id);
+                assert_eq!(bob_swap_params.peer.peer_id, alice_peer_id);
+                assert_eq!(alice_local_swap_id, alice_node_id);
+                assert_eq!(bob_local_swap_id, bob_node_id);
+                assert_eq!(alice_secret_hash, bob_secret_hash);
```

Comment:
We should check on the identities as well, if it is necessary to derive `Partial_Eq` for that I would say it's OK. @thomaseizinger @tcharding what do you think?

Reply:
I don't think its worth checking the identities, we already check that this all works in the unit tests.  This test is just checking we can glue everything together correctly.